### PR TITLE
olsrd: switch to procd

### PR DESCRIPTION
**Pull request by PolynomialDivision**
Run-Tested: Fritzbox 7530 (ipq40xx)

### CODE REVIEW

File: olsrd/files/olsrd4.init [hunk 1]
```diff
@@ -1,14 +1,34 @@
...
 
 START=65
+USE_PROCD=1
+BIN=/usr/sbin/olsrd
+OLSRD=olsrd
+CONF=/var/etc/$OLSRD.conf
 
-SERVICE_DAEMONIZE=1
-SERVICE_WRITE_PID=1
+tart_service() {
```
Comment (jow-) [inline olsrd/files/olsrd4.init:12]: Typo (missing `s`)

File: olsrd/files/olsrd4.init [hunk 2]
```diff
@@ -1,14 +1,34 @@
...
+	olsrd_generate_config $OLSRD
+    procd_set_param command "$BIN"
```
Comment (jow-) [inline olsrd/files/olsrd4.init:14]: Identation is off (tab vs. space?)

File: olsrd/files/olsrd4.init [hunk 3]
```diff
@@ -1,14 +1,34 @@
...
+
+	olsrd_setup_smartgw_rules
+}
+
+service_triggers() {
+	# restart on ifup interface events
+	for n in $(olsrd_list_configured_interfaces $OLSRD); do
+		procd_add_reload_interface_trigger $n /etc/init.d/$OLSRD reload
+	done
+}
```
Comment (jow-) [inline olsrd/files/olsrd4.init:34]: Missing final newline

File: olsrd/files/olsrd6.init [hunk 4]
```diff
@@ -1,14 +1,37 @@
...
+
+	olsrd_setup_smartgw_rules
+}
+
+service_triggers() {
+	# restart on ifup interface events
+	for n in $(olsrd_list_configured_interfaces $OLSRD); do
+		procd_add_reload_interface_trigger $n /etc/init.d/$OLSRD reload
+	done
+}
```
Comment (jow-) [inline olsrd/files/olsrd6.init:37]: Missing final newline

File: olsrd/files/olsrd.sh [hunk 5]
```diff
@@ -1,10 +1,5 @@
 #!/bin/sh /etc/rc.common
-# Copyright (C) 2008-2017 OpenWrt.org
-
-START=65
-
-SERVICE_DAEMONIZE=1
-SERVICE_WRITE_PID=1
+# Copyright (C) 2008-2021 OpenWrt.org
```
Comment (jow-) [inline olsrd/files/olsrd.sh:2]: Please leave the original copyright statements alone. Either add your own (Copyright (C) 2021 Nick Hainke <vincent@systemli.org>) or don't do anything about it. The same applies for other instances of the OpenWrt copyright.

Comment (PolynomialDivision) [reply olsrd/files/olsrd.sh]: Okay, thanks for clarifying.

Comment (PolynomialDivision) [reply olsrd/files/olsrd.sh]: changing to spdx is okay?

**Comment by PolynomialDivision** [comment]
Starting olsrd with it, will hold olsrd in the foreground?

**Comment by PolynomialDivision** [comment]
@jow- Anything else? Can I merge?

**Comment by PolynomialDivision** [comment]
@andibraeu

**Pull request merged**